tests/server/api/routers/v1/test_datasets.py

```diff
@@ -315,3 +320,111 @@ async def test_get_dataset_download_specific_version(test_client, dataset_with_r
         )
     ).sort_index(axis=1)
     assert_frame_equal(actual, expected)
+
+
+async def test_post_dataset_upload_csv(test_client, session):
+    name = str(random())
+    file = gzip.compress(b"a,b,c,d,e,f\n1,2,3,4,5,6\n")
+    response = await test_client.post(
+        url="v1/datasets/upload?sync=true",
+        files={"file": (" ", file, "text/csv", {"Content-Encoding": "gzip"})},
+        data={
+            "action": "create",
+            "name": name,
+            "input_keys[]": ["a", "b"],
+            "output_keys[]": ["c", "d"],
+            "metadata_keys[]": ["e", "f"],
+        },
+    )
+    assert response.status_code == 200
+    assert (dataset_id := response.json().get("dataset_id"))
+    del response, file
```

Comment:
Can we separate into distinct test cases? Having them in a single test case makes it more difficult to debug the test case if something fails IMO. These also test distinct functionality, one for create and one for append.

Reply:
You're asking for a separate (stand-alone) test for creation only, right? The append test has to have a creation step regardless. Then I'm not sure how much more an extra stand-alone test for creation is buying us since it's just duplicated coverage. Or are you saying we need to have a fixture that already has records and test only the append step? Either way, I don't think it's worth the extra effort/complication here.

Reply:
I think you might be right. Sounds good.